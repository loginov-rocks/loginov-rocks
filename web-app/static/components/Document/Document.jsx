import PropTypes from 'prop-types';
import * as React from 'react';

export const Document = ({
  Body, children, Head, Html,
}) => (
  <Html>
    <Head>
      <meta charset="utf-8" />
      <meta content="Personal Website of Danila Loginov" name="description" />
      <meta content="all" name="robots" />
      <meta content="width=device-width, initial-scale=1" name="viewport" />
      <title>Danila Loginov</title>
      <link href="/manifest.json" rel="manifest" />
    </Head>
    <Body>
      {children}
    </Body>
  </Html>
);

Document.propTypes = {
  Body: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
  ]).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
    PropTypes.array,
  ]).isRequired,
  Head: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
  ]).isRequired,
  Html: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
  ]).isRequired,
};
