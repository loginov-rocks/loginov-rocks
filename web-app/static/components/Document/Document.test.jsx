import { shallow } from 'enzyme';
import PropTypes from 'prop-types';
import * as React from 'react';

import { Document } from './Document';

const MockComponent = ({ children }) => <div>{children || null}</div>;

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
    PropTypes.array,
    PropTypes.string,
  ]).isRequired,
};

it('matches snapshot', () => {
  const wrapper = shallow(
    <Document Body={MockComponent} Head={MockComponent} Html={MockComponent}>
      <MockComponent>
        Hello, world!
      </MockComponent>
    </Document>,
  );

  expect(wrapper).toMatchSnapshot();
});
