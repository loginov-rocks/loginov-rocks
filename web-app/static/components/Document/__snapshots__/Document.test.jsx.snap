// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<MockComponent>
  <MockComponent>
    <meta
      charset="utf-8"
    />
    <meta
      content="Personal Website of Danila Loginov"
      name="description"
    />
    <meta
      content="all"
      name="robots"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
    <title>
      Danila Loginov
    </title>
    <link
      href="/manifest.json"
      rel="manifest"
    />
  </MockComponent>
  <MockComponent>
    <MockComponent>
      Hello, world!
    </MockComponent>
  </MockComponent>
</MockComponent>
`;
